src/main/java/core/basesyntax/Square.java

```diff
@@ -0,0 +1,21 @@
+package core.basesyntax;
+
+public class Square extends Figure {
+    private double sideLength;
+
+    public Square(Color color, double sideLength) {
+        super(color);
+        this.sideLength = sideLength;
+    }
+
+    @Override
+    public double getArea() {
+        return sideLength * sideLength;
+    }
+
+    @Override
+    public void print() {
+        System.out.println("Figure: square, area: " + getArea() + " sq. units, side length: "
+                + sideLength + " units, color: " + new ColorSupplier().getRandomColor());

```

Comment:
Avoid creating a new instance of ColorSupplier in the print method. This should be passed to the constructor of the figure or stored as a class-level variable if needed.